src/Symfony/Component/Messenger/Transport/Receiver/FailSilentlyReceiver.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Receiver;
+
+/**
+ * @author Mathias STRASSER <contact@roukmoute.fr>
+ *
+ * @experimental in 4.2
+ */
+final class FailSilentlyReceiver implements ReceiverInterface
+{
+    /** @var ReceiverInterface */
+    private $decoratedReceiver;
+
+    public function __construct(ReceiverInterface $decoratedReceiver)
+    {
+        $this->decoratedReceiver = $decoratedReceiver;
+    }
+
+    public function receive(callable $handler): void
+    {
+        $this->decoratedReceiver->receive(
+            function ($message) use ($handler) {
+                try {
+                    $handler($message);
+                } catch (\Throwable $exception) {
```

Comment:
An exception thrown by a message handler is a good indicator of there being something that needs addressing. Silently swallowing such exceptions hides potential faults in the relevant message handler.

Can we/should we introduce an option to log the exception?

Reply:
Or, to put it another way, I'd love to be able to run `messenger:consume-messages` knowing that it won't stop if an exception occurs within a message handler. But if an exception does occur, I would really need to know about it as it probably relates to a bug that I need to fix.